youtube_dl/extractor/senateisvp.py

```diff
@@ -5,149 +5,169 @@
+        },
     }]
 
     @staticmethod
+    # returns url from an iframe
     def _search_iframe_url(webpage):
         mobj = re.search(
-            r"<iframe[^>]+src=['\"](?P<url>https?://www\.senate\.gov/isvp/?\?[^'\"]+)['\"]",
+            r'''<iframe\b[^>]+\bsrc\s*=\s*(['"])(?P<url>https?://www\.senate\.gov/isvp/?\?(?:(?!\1)\S)+)''',
             webpage)
         if mobj:
             return mobj.group('url')
 
+    # returns stream_number, stream_domain, stream_id, and msl3
     def _get_info_for_comm(self, committee):
-        for entry in self._COMM_MAP:
-            if entry[0] == committee:
-                return entry[1:]
+        return self._COMM_MAP[committee][0:]
 
     def _real_extract(self, url):
+        # smuggled_data may contain a forced title that should be used
         url, smuggled_data = unsmuggle_url(url, {})
-
         qs = compat_parse_qs(re.match(self._VALID_URL, url).group('qs'))
-        if not qs.get('filename') or not qs.get('type') or not qs.get('comm'):
-            raise ExtractorError('Invalid URL', expected=True)
 
-        video_id = re.sub(r'.mp4$', '', qs['filename'][0])
+        # error handling for invalid URL - specify what
+        if not qs.get('comm'):
+            raise ExtractorError('''Invalid URL. Missing committee in query parameters.''', expected=True)
+        if not qs.get('filename'):
+            raise ExtractorError('''Invalid URL. Missing filename in query parameters.''', expected=True)
 
-        webpage = self._download_webpage(url, video_id)
+        committee = qs.get('comm')[0]
+        filename = qs.get('filename')[0]
+        video_id = re.sub(r'\.mp4$', '', filename)
 
-        if smuggled_data.get('force_title'):
-            title = smuggled_data['force_title']
-        else:
-            title = self._html_search_regex(r'<title>([^<]+)</title>', webpage, video_id)
-        poster = qs.get('poster')
-        thumbnail = poster[0] if poster else None
+        # there is no point in pulling the title from the webpage since it always defaults to 'Integrated Senate Player'
+        title = smuggled_data.get('force_title') or filename
+        stream_number, stream_domain, stream_id, msl3 = self._get_info_for_comm(committee)
 
-        video_type = qs['type'][0]
-        committee = video_type if video_type == 'arch' else qs['comm'][0]
-        stream_num, domain = self._get_info_for_comm(committee)
+        # the possible locations that the video could be contained at.
+        possible_manifest_urls = [
+            'https://www-senate-gov-media-srs.akamaized.net/hls/live/%d/%s/%s/master.m3u8' % (stream_id, committee, filename),
+            'https://www-senate-gov-msl3archive.akamaized.net/%s/%s_1/master.m3u8' % (msl3, filename),
+            '{stream_domain}/i/%s_1@%d/master.m3u8' % (filename, stream_number),
+            'https://ussenate-f.akamaihd.net/i/%s' % video_id,
```

Comment:
The page JS is using this (actually probably not since the page code is wrong), supposing that the part before `/i/` is the stream format for the `arch` row of the map:
```suggestion
            'https://ussenate-f.akamaihd.net/i/%s.mp4/master.m3u8' % video_id,
```
Or extract it from the page map:
```py
        _, stream_arch, _, _ = self._get_info_for_comm('arch')
...
            '%s/i/%s.mp4/master.m3u8' % (stream_arch, video_id),
```